controllers/actions-controller.js

```diff
@@ -0,0 +1,93 @@
+const db = require('../models');
+
+module.exports = (app) => {
+
+  // Show action
+  app.get('/actions/:id', (req, res) => {
+    const actionId = req.params.id
+    db.Action.findById(actionId)
+    .then((action) => {
+      console.log("Response from action/Show: ", action)
+      res.status(200)
+      res.json({
+        message: "action request successful",
+        action
+      })
+    })
+    .catch((err) => {
+      console.log(err);
+      res.status(400);
+      res.json({
+        message: "Error!",
+        error: err
+      })
+    })
+  });
+
+  //TODO: Code Review  whether the actionId exists, it does not ever hits catch error
+    // UPDATE a action
+  app.put('/actions/:id/update', (req, res) => {
+    const actionId = req.params.id
+    const action = req.body
+    db.Action.update(action, {
+      where: { id: actionId }
+    }).then((response) => {
+        res.status(200)
+        res.json({
+          msg: 'action updated successfully!',
+        })
+      }).catch((err) => {
+        console.log(err);
+        res.status(400);
+        res.json({
+          message: "Error!",
+          error: err
+        })
+      })
+    });
+
+  // Create a action
+  app.post('/criteria/:criterionId/actions/create', (req, res) => {
+    const newAction = {...req.body, criterionId: req.params.criterionId}
+    db.Action.create(newAction)
+    .then((action) => {
+      console.log("Response from action/Create: ", action)
+      res.status(200)
+      res.json({
+        msg: 'action added successfully!',
+        action
+      })
+    })
+    .catch((err) => {
+      console.log(err);
+      res.status(400);
+      res.json({
+        message: "Error!",
+        error: err
+      })
+    })
+  });
+
+  //TODO: Code Review  whether the actionId exists, it does not ever hits catch error
```

Comment:
I'm not sure I understand this comment...

Reply:
If you trigger this route without an ID, what happens?

Reply:
I think @Jeffchiucp is thinking that there is an `actionId` somewhere.  There isn't.  The `criterionId` is stored inside the `Actions` table since it's a one to many relationship between `Criteria` and `Actions`.